lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -907,7 +907,7 @@ public HeldLocksGrant convertToGrant(HeldLocksToken token) {
     @Override
     public HeldLocksToken useGrant(LockClient client, HeldLocksGrant grant) {
         com.palantir.logsafe.Preconditions.checkNotNull(client);
-        com.palantir.logsafe.Preconditions.checkArgument(client != INTERNAL_LOCK_GRANT_CLIENT);
+        com.palantir.logsafe.Preconditions.checkArgument(!INTERNAL_LOCK_GRANT_CLIENT.equals(client));
```

Comment:
It looks like the difference here is that you'll return false if the `clientId`s are different (where before, if I understand this correctly, just if the references are the same). Presumably the `clientId` is supposed to uniquely identify a client, but I haven't dug into the code to verify this.